protocols/dcutr/examples/client.rs

```diff
@@ -0,0 +1,178 @@
+use std::task::{Context, Poll};
+
+fn main() -> Result<(), Box<dyn Error>> {
+    env_logger::init();
+
+    let local_key = generate_ed25519(std::env::args().nth(1).unwrap().parse().unwrap());
+    let local_peer_id = PeerId::from(local_key.public());
+    println!("Local peer id: {:?}", local_peer_id);
+
+
+    let (transport, client) = Client::new_transport_and_behaviour(local_peer_id, block_on(DnsConfig::system( TcpConfig::new().port_reuse(true))).unwrap());
+
+    let noise_keys = noise::Keypair::<noise::X25519Spec>::new()
+        .into_authentic(&local_key)
+        .expect("Signing libp2p-noise static DH keypair failed.");
+
+    let transport = transport
+        .upgrade()
+        .authenticate_with_version(noise::NoiseConfig::xx(noise_keys).into_authenticated(), upgrade::AuthenticationVersion::V1SimultaneousOpen)
+        .multiplex(libp2p_yamux::YamuxConfig::default())
+        .boxed();
+
+    #[derive(NetworkBehaviour)]
+    #[behaviour(out_event = "Event", event_process = false)]
+    struct Behaviour {
+        relay_client: Client,
+        ping: Ping,
+        identify: Identify,
+        dcutr: dcutr::behaviour::Behaviour,
+    }
+
+    #[derive(Debug)]
+    enum Event {
+        Ping(PingEvent),
+        Identify(IdentifyEvent),
+        Relay(client::Event),
+        Dcutr(dcutr::behaviour::Event),
+    }
+
+    impl From<PingEvent> for Event {
+        fn from(e: PingEvent) -> Self {
+            Event::Ping(e)
+        }
+    }
+
+    impl From<IdentifyEvent> for Event {
+        fn from(e: IdentifyEvent) -> Self {
+            Event::Identify(e)
+        }
+    }
+
+    impl From<client::Event> for Event {
+        fn from(e: client::Event) -> Self {
+            Event::Relay(e)
+        }
+    }
+
+    impl From<dcutr::behaviour::Event> for Event {
+        fn from(e: dcutr::behaviour::Event) -> Self {
+            Event::Dcutr(e)
+        }
+    }
+
+    let behaviour = Behaviour {
+        relay_client: client,
+        ping: Ping::new(PingConfig::new()),
+        identify: Identify::new(IdentifyConfig::new(
+            "/TODO/0.0.1".to_string(),
+            local_key.public(),
+        )),
+        dcutr: dcutr::behaviour::Behaviour::new(),
+    };
+
+    let mut swarm = Swarm::new(transport, behaviour, local_peer_id);
+
+    if let (Some(external_ip), Some(port)) = (std::env::args().nth(2), std::env::args().nth(3)) {
+        let listen_addr = Multiaddr::empty()
+            .with("0.0.0.0".parse::<Ipv4Addr>().unwrap().into())
+            .with(Protocol::Tcp(port.parse().unwrap()));
+        swarm.listen_on(listen_addr)?;
+
+        let external_addr = Multiaddr::empty()
+            .with(external_ip.parse::<Ipv4Addr>().unwrap().into())
+            .with(Protocol::Tcp(port.parse().unwrap()));
+        swarm.add_external_address(external_addr, AddressScore::Infinite);
+    } else {
+        panic!();
+    }
+
+    if let (Some(mode), Some(addr)) = (std::env::args().nth(4), std::env::args().nth(5)) {
```

Comment:
Maybe we can change to structopt with description for each parameter as we did in the relay_v1
and then we can write some example of 3 nodes settings together?

Reply:
Yes for sure, that would be nice. This is a hack at most at the moment. Want to file a pull request against my branch on my fork?

Reply:
Yes, I will do.